templates/ovncontroller/bin/functions

```diff
@@ -57,34 +57,77 @@ function configure_external_ids {
     fi
 }
 
+# Returns the set difference between $1 and $2
+function set_difference {
+    echo "$(comm -23 <(sort <(echo $1 | xargs -n1)) <(sort <(echo $2 | xargs -n1)))"
+}
+
 # Configure bridge mappings and physical bridges
 function configure_physical_networks {
     local OvnBridgeMappings=""
+    local br_new=""
     for physicalNetwork in ${PhysicalNetworks}; do
         br_name="br-${physicalNetwork}"
-        ovs-vsctl --may-exist add-br ${br_name}
-        ovs-vsctl --may-exist add-port ${br_name} ${physicalNetwork}
         bridgeMapping="${physicalNetwork}:${br_name}"
         if [ -z "$OvnBridgeMappings"]; then
             OvnBridgeMappings=$bridgeMapping
+            br_new=$br_name
         else
             OvnBridgeMappings="${OvnBridgeMappings},${bridgeMapping}"
+            br_new="${br_new} ${br_name}"
+        fi
+    done
+
+    # Current configured bridges.
+    ovn_bms=$(ovs-vsctl --if-exists get open . external_ids:ovn-bridge-mappings|tr -d '"')
+    local br_current=""
+    for bm in ${ovn_bms/,/ }; do
+        if [ -z "$br_current"]; then
+            br_current=${bm##*:}
+        else
+            br_current="${br_current} ${bm##*:}"
         fi
     done
+
+    # Bridges to add and delete.
+    local br_to_delete=""
+    local br_to_add=""
+    br_to_delete=$(set_difference "$br_current" "$br_new")
+    br_to_add=$(set_difference "$br_new" "$br_current")
+
+    # Add the new bridges.
+    for br_name in ${br_to_add}; do
+        ovs-vsctl --may-exist add-br ${br_name}
+        ovs-vsctl --may-exist add-port ${br_name} ${br_name##*-}
+    done
+
+    # Delete the old bridges not longer present in "OvnBridgeMappings" and the
+    # patch ports in "br-int".
+    for br_name in ${br_to_delete}; do
+        # List all ports "type=patch", retrieve the "options:peer" parameter
+        # and delete the peer port. The "$port" will be deleted during the
+        # bridge deletion.
+        #   $ ovs-vsctl --column=options --bare find interface name=patch-provnet1-to-br-int type=patch
+        #     peer=patch-br-int-to-provnet1
+        for port in $(ovs-vsctl list-port ${br_name}); do
+            options=$(ovs-vsctl --column=options --bare find interface ${port} type=patch)
+            for option in ${options}; do
+                key=$(echo $option | cut -f1 -d "=")
+                peer_name=$(echo $option | cut -f2 -d "=")
+                if [[ "$key" == "peer" ]]; then
+                    ovs-vsctl --if-exists del-port br-int $port ${peer_name}
+                fi
+            done
+        done
+
+        ovs-vsctl --if-exists del-br ${br_name}
```

Comment:
That's a mistake when I pushed the code. This is not the latest code I locally have. This code will fail when:
* The bridge to delete doesn't exist (L112)
* There are no patch ports in the bridge (L113)

This is the code I changed, including these conditions:
```
        br="$(ovs-vsctl list-br | grep ${br_name})"
        if [ -n "$br" ]; then
            for port in $(ovs-vsctl list-ports ${br_name}); do
                port_type=$(ovs-vsctl --column=type --bare list interface ${port})
                if [ "$port_type" == "patch" ]; then
                  options=$(ovs-vsctl --column=options --bare list interface ${port})
                  for option in ${options}; do
                      key=$(echo $option | cut -f1 -d "=")
                      peer_name=$(echo $option | cut -f2 -d "=")
                      if [ "$key" == "peer" ]; then
                          ovs-vsctl --if-exists del-port br-int $port ${peer_name}
                      fi
                  done
                fi
            done

            ovs-vsctl --if-exists del-br ${br_name}
        fi
```

Reply:
okk thanks will update, including simplified version.